### ISSUE TITLE
Question: How to terminate any robocopy.exe processes when disposing a RoboCommand

### ISSUE THREAD

**Issue by vittoso** [tjscience/RoboSharp#171]
Hi,
this is a question more than an issue.
Is there a way to force termination on any running robocopy.exe process when the corresponding RoboCommand is Disposed/Finalized or destroyed?

At the moment if I terminate my application (.NET Framework) the underlying robocopy.exe keeps running.

I couldn't find any info about this in the Wiki, so I wonder whether this behaviour is possible.

Thanks

**Comment by PCAssistSoftware** [comment]
You just need to ensure you dispose of RoboCommand correctly when your job ends or when your application closes if a Form application then handle FormClosing or similar.

So for example in VB.net if you had declared using "Dim job As New RoboCommand" then when job finished or app closing you can dispose using job.dispose which will then kill any running RoboCopy process as long as StopIfDisposing is set to True



![image](https://user-images.githubusercontent.com/10872799/236880139-218736f3-7f4a-480a-abba-2c834a0dbac7.png)

**Comment by vittoso** [comment]
Hi, thank you for your answer. 
I am already calling Dispose() (I'm using C# so I'm wrapping it in using statement and it works fine, but if the application is killed (or I stop debugging the application in Visual Studio), Dispose is not called hence the robocopy.exe subprocess keeps running.

I understand that maybe there is no way to handle this case, as the process that could dispose and clear the resource is killed so it cannot run any code.

**Comment by PCAssistSoftware** [comment]
I would just add some code in to the app closing code perhaps to check for and kill and running robocopy.exe processes perhaps? as that is similar to what the bult in methods would do

**Comment by PCAssistSoftware** [comment]
Closed as no further responses - OP can reopen if required

**Comment by vittoso** [comment]
Just a quick not to share how I solved it, in case it is useful for anybody else, the issue can remain closed as there's no change to do in RoboSharp.
I created a Win32 JobObject configuring it to kill all child processes (i.e. robocopy.exe) when the root process is killed. I was inspired by the approach listed in https://www.meziantou.net/killing-all-child-processes-when-the-parent-exits-job-object.htm with a few changes as I was in an environment where I cannot use source generators.
